### ISSUE TITLE
no-value-for-parameter false positive with sqlalchemy.ext.hybrid.hybrid_method

### ISSUE THREAD

**Issue by tamird** [pylint-dev/pylint#9757]
### Bug description

Taking an example from the [`sqlalchemy.ext.hybrid.hybrid_method` docs](https://docs.sqlalchemy.org/en/20/orm/extensions/hybrid.html#sqlalchemy.ext.hybrid.hybrid_method):

```python
from __future__ import annotations

from sqlalchemy.ext.hybrid import hybrid_method
from sqlalchemy.ext.hybrid import hybrid_property
from sqlalchemy.orm import DeclarativeBase
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column


class Base(DeclarativeBase):
    pass

class Interval(Base):
    __tablename__ = 'interval'

    id: Mapped[int] = mapped_column(primary_key=True)
    start: Mapped[int]
    end: Mapped[int]

    def __init__(self, start: int, end: int):
        self.start = start
        self.end = end

    @hybrid_property
    def length(self) -> int:
        return self.end - self.start

Interval.length() # <--- false positive here
```

`Interval.length` is both a class method and an instance method, and can be called as a class method when building SQL queries. pylint reports no-value-for-parameter on this usage.

### Command used

```shell
pylint foo.py
```

### Pylint output

```python
E1120: No value for argument 'self' in unbound method call (no-value-for-parameter)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 3.2.3
astroid 3.2.2
Python 3.12.4 (main, Jun 24 2024, 06:23:58) [Clang 15.0.0 (clang-1500.3.9.4)]
```

**Comment by Pierre-Sassoulas** [comment]
I cannot reproduce this, could you give your version of sqlalchemy ? Do you have a specific configuration ? I tested pylint 3.2.3, astroid 3.2.2, CPython 3.12, across SQLAlchemy 2.0.0, 2.0.15 and 2.0.31,

**Comment by tamird** [comment]
Yikes, I filed this issue over 2 years ago at a prior job. I'm afraid I can't recover any additional details. I was likely using whatever version of sqlalchemy was latest at that time.

**Comment by Pierre-Sassoulas** [comment]
Yeah, we don't have maintainer using sqlalchimy and it shows 😅 That would be [2.0.31](https://pypi.org/project/SQLAlchemy/2.0.31/) from june 18th 2024 that I tested. I'm going to close, if anyone still encounter the issue please reopen with a new reproducing snippet.
